service/controller/clusterapi/v21/resource/workercount/create.go

```diff
@@ -57,20 +56,32 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 		nodes = l.Items
 
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found nodes of tenant cluster %#q", key.ClusterID(&cr)))
+	}
+
+	{
+
 		for _, n := range nodes {
+			id := n.Labels[label.MachineDeployment]
+
+			if cc.Status.Worker == nil {
+				cc.Status.Worker = map[string]controllercontext.ContextStatusWorker{}
+			}
+
+			{
+				val := cc.Status.Worker[id]
+				val.Nodes++
+				cc.Status.Worker[id] = val
```

Comment:
Why just `cc.Status.Worker[id].Nodes++` would not work?

Reply:
https://stackoverflow.com/questions/32751537/why-do-i-get-a-cannot-assign-error-when-setting-value-to-a-struct-as-a-value-i